src/Microsoft.ML.Data/Scorers/FeatureContributionCalculatingTransformer.cs

```diff
@@ -0,0 +1,673 @@
+        /// computes model-specific contribution scores for each feature.
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="predictor">A trained model that supports Feature Contribution Calculation, and which will be used for scoring.</param>
+        /// <param name="featureColumn">The name of the feature column that will be used as input.</param>
+        /// <param name="top">The number of top contributing features for each data sample that will be retained in the FeatureContribution column.</param>
+        /// <param name="bottom">The number of least contributing features for each data sample that will be retained in the FeatureContribution column.</param>
+        /// <param name="normalize">Whether the feature contributions should be normalized.</param>
+        /// <param name="stringify">Whether to output feature contributions in string key-value format.</param>
+        public FeatureContributionCalculatingTransformer(IHostEnvironment env, IFeatureContributionMapper predictor,
+            string featureColumn = DefaultColumnNames.Features,
+            int top = FeatureContributionCalculatingEstimator.Defaults.Top,
+            int bottom = FeatureContributionCalculatingEstimator.Defaults.Bottom,
+            bool normalize = FeatureContributionCalculatingEstimator.Defaults.Normalize,
+            bool stringify = FeatureContributionCalculatingEstimator.Defaults.Stringify)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(FeatureContributionCalculatingTransformer)))
+        {
+            // Other Parameters are checked in the constructor of the BindableMapper.
+            Host.CheckValue(predictor, nameof(predictor));
+            Host.CheckNonEmpty(featureColumn, nameof(featureColumn));
+
+            _featureColumn = featureColumn;
+            _mapper = new BindableMapper(Host, predictor, top, bottom, normalize, stringify);
+        }
+
+        // Factory method for SignatureLoadModel
+        private FeatureContributionCalculatingTransformer(IHostEnvironment env, ModelLoadContext ctx)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(FeatureContributionCalculatingTransformer)))
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            // *** Binary format ***
+            // string features
+            // BindableMapper mapper
+
+            _featureColumn = ctx.LoadNonEmptyString();
+            ctx.LoadModel<BindableMapper, SignatureLoadModel>(env, out _mapper, ModelFileUtils.DirPredictor);
+        }
+
+        // Factory method for SignatureLoadRowMapper.
+        internal static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => new FeatureContributionCalculatingTransformer(env, ctx).MakeRowMapper(Schema.Create(inputSchema));
```

Comment:
do you need both this and the ctor?

Reply:
They do different things, one loads a IRowMapper, the other one loads a FeatureContributionCalculatingTransformer

---
In reply to: [240767728](https://github.com/dotnet/machinelearning/pull/1847#discussion_r240767728) [](ancestors = 240767728)

Reply:
All our transformers have both, but I am not entirely sure when you need one or the other.

---
In reply to: [241136913](https://github.com/dotnet/machinelearning/pull/1847#discussion_r241136913) [](ancestors = 241136913,240767728)